runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/npc/Rs2Npc.java

```diff
@@ -154,6 +155,16 @@ public static Optional<NPC> getNpc(int id, List<Integer> excludedIndexes) {
                         value.getLocalLocation().distanceTo(Microbot.getClient().getLocalPlayer().getLocalLocation())));
     }
 
+    public static Optional<NPC> getNPCNearestToPlayer() {
```

Comment:
Does the following method not qualify?

`    public static Stream<NPC> getNpcs() {
        Stream<NPC> npcs = Microbot.getClientThread().runOnClientThread(() -> Microbot.getClient().getNpcs().stream()
                .filter(x -> x != null && x.getName() != null && !x.isDead())
                .sorted(Comparator.comparingInt(value -> value.getLocalLocation()
                        .distanceTo(Microbot.getClient().getLocalPlayer().getLocalLocation()))));`

Reply:
The getNpcs() also sorts from closest npc to further away. So if you get the first element of the stream, should always have the closest npc

Reply:
I should stop working on this stuff so late. I have simplified the logic for this in commit c6202bb